src/layout/Summary2/SummaryComponent2/ComponentSummary.tsx

```diff
@@ -11,54 +11,45 @@ import type { LayoutNode } from 'src/utils/layout/LayoutNode';
 
 interface ComponentSummaryProps {
   componentNode: LayoutNode;
-  summaryOverrides: CompSummary2Internal['overWriteProperties'];
+  summaryOverrides?: CompSummary2Internal['overrides'];
 }
 
 interface ResolveComponentProps {
   summaryProps: CompSummary2External;
-  summaryOverrides: any;
+  summaryOverrides?: CompSummary2Internal['overrides'];
 }
-
 export function ComponentSummary({ componentNode, summaryOverrides }: ComponentSummaryProps) {
-  if (componentNode.isHidden()) {
-    return null;
-  }
-
-  const overrides = summaryOverrides?.find((override) => override.componentId === componentNode.item.id);
-
-  const childComponents =
-    componentNode.item.type === 'Group' &&
-    componentNode.item.childComponents.map((child) => (
-      <ComponentSummary
-        componentNode={child}
-        key={child.item.id}
-        summaryOverrides={summaryOverrides}
-      />
-    ));
+  const override = summaryOverrides?.find((override) => override.componentId === componentNode.item.id);
 
   const renderedComponent = componentNode.def.renderSummary2
-    ? componentNode.def.renderSummary2(componentNode as LayoutNode<any>, overrides)
+    ? componentNode.def.renderSummary2(componentNode as LayoutNode<any>, override)
     : null;
 
   if (!renderedComponent) {
     return null;
   }
 
+  if (override?.hidden) {
+    return null;
+  }
+
   return (
     <Grid
       item={true}
       className={cn(pageBreakStyles(componentNode.item?.pageBreak), classes.summaryItem)}
       {...gridBreakpoints(componentNode.item.grid)}
     >
       {renderedComponent}
-      {childComponents}
     </Grid>
   );
 }
 
 export function ResolveComponent({ summaryProps, summaryOverrides }: ResolveComponentProps) {
-  const resolvedComponent = useNode(summaryProps.whatToRender.id);
+  if (!summaryProps.target?.id) {
+    throw new Error('Tried to render component without, this should not happen.');
+  }
```

Comment:
Without what? :wink: Also, isn't this a configuration error by the app developer? In such cases we should log to DevTools instead, via `window.logError()`.